crates/nu-cli/src/commands/math/median.rs

```diff
@@ -129,7 +141,39 @@ fn compute_average(values: &[Value], name: impl Into<Tag>) -> Result<Value, Shel
         )
     })?;
     let total_rows = UntaggedValue::decimal(number);
-    let total = sum(UntaggedValue::int(0).into_untagged_value(), values.to_vec())?;
+
+    let are_bytes = values
+        .get(0)
+        .ok_or_else(|| {
+            ShellError::unexpected("Cannot perform aggregate math operation on empty data")
+        })?
+        .is_filesize();
+
+    let total = if are_bytes {
+        to_byte(&sum(
+            UntaggedValue::int(0).into_untagged_value(),
+            values
```

Comment:
Just so I understand this a bit better - is the idea here to do all the math using just integer values, and then to convert that back to a filesize if we know we're in filesize mode?

Something I was looking for but wasn't sure about - does this also allow you to mix filesizes and raw ints? We might want to lock it down so once we see one filesize, we require that they're all filesizes

Reply:
Hi @jonathandturner!
Yes, I think that is how the summation reducer is currently being used in other subcommands that are are using it, (such as `math sum` or `math avg`). They are checking if the first value is filesize, if the first value is a filesize type, the code iterates over the values, and then converts filesize types to integers. performs the sum, and then converts that summed value back to a filesize. This conversion to int, and then summing and converting back to filesize, wasn't happening in the median command, which is why like `echo [10kb, 500kb] | math median` was failing.

I _think_ this is because the way the summation reducer is set up, it starts with with an initial integer value of zero, and then reduces the values looping over all of them. 
https://github.com/nushell/nushell/blob/c012d648fb23da92d07bb7c25d3531ab7d6d8078/crates/nu-cli/src/commands/math/reducers.rs#L62-L75

Looking through the `compute_values` function, there is no functionality in that function for adding an int, and a filesize.

Thinking through this more though, I wonder if rather than having all of these conversions happening in at the subcommand level (converting to int, summing, and then converting to filesize) it would be better to have the reducer just set up to initialize the accumulator value, with the same type as the contents of the input data.

Also, good point about the the mix of filesize and raw ints... I think it depends on if the first value is an int or a filesize. If the first value in the data is an int, no conversion will happen, and the summation will fail if other values in the data are a filesize. If the first value is a filesize, it will convert all filesive values to int, and this would work.  
For example, this would work:
```shell
❯ echo [1kb, 5000] | math median
───┬────────
 0 │ 3.0 KB 
───┴────────
```
But this would fail...  
```shell
❯ echo [1, 5000kb] | math median
error: Coercion error
  ┌─ shell:1:1
  │
1 │ echo [10kb, 500kb] | math median
  │ ^
  │ │
  │ integer
  │ filesize(in bytes)
```
Same thing holds for how the sum subcommand is currently coded: `echo [1kb, 5000] | math sum` currently works, while `echo [1, 5000kb] | math sum` fails.

Sorry for the long response, but I guess after thinking through this more, maybe this isn't the correct fix?

Reply:
I suspect what you found is where the logic is also wrong in the other commands. Your investigation makes sense to me, seems like we're a little too loose, and we should tighten things a bit. We can start with median, and then make some issue to help track down the looseness in the other commands.

If you're up for it, I'd say that what you suggest with setting the type by the first input makes sense.

Reply:
Ok, I gave this a go. Nothing too crazy, essentially I reverted the median code back to the way it was, and then simply changed the summation reducer logic. Now instead of defaulting to an initial value of a zero int, it chooses either filesize if the first value is a filesize type, otherwise it will use an integer. Because the other commands that are using the summation reducer are already converting the input values to an integer prior to summing, there is no impact to any other function.

This also serves to as a check for if the data into the median command is mixing filesize types and raw numeric values. 
For instance now both `echo [10, 100kb] | math median` and `echo [10kb, 100] | math median` will both error, where before this change latter would have returned a result.

Additionally I pulled out the branch of the summation reducer in the `reducer_for` function, before it was getting passed through the `formula` function, but because all of the initial values were being set to zero, this had no impact to the reducer. The `formula` function was multiplying the input initial value, but because not all value types support multiplication (filesize for instance) this just limited the value types that the summation reducer could use. Again pulling it into it's own branch in the match `reducer_for(` match statement has no impact on the summation reducer.